smart-contract-verifier/src/vyper/compiler.rs

```diff
@@ -0,0 +1,190 @@
+use std::path::Path;
+
+use ethers_solc::{error::SolcError, CompilerInput, CompilerOutput, Solc};
+
+use crate::compiler::{self, EvmCompilerAgent};
+
+pub struct VyperCompilerAgent {}
+
+impl VyperCompilerAgent {
+    pub fn new() -> Self {
+        VyperCompilerAgent {}
+    }
+}
+
+impl EvmCompilerAgent for VyperCompilerAgent {
+    fn compile(
+        &self,
+        path: &Path,
+        ver: &compiler::Version,
+        input: &CompilerInput,
+    ) -> Result<CompilerOutput, SolcError> {
+        let _ = ver;
+        let vyper_output: types::VyperCompilerOutput = Solc::from(path).compile_as(input)?;
+        Ok(CompilerOutput::from(vyper_output))
+    }
+}
+
+mod types {
+    use std::collections::BTreeMap;
+
+    use ethers_solc::{
+        artifacts::{Contract, Severity},
+        CompilerOutput,
+    };
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    pub struct SourceLocation {
+        pub file: String,
+    }
```

Comment:
What about `lineno` and `col_offset` fields, and `SourceLocation` inside `VyperError` struct?

Reply:
why do we need them?

Reply:
It would be useful for users if we could provide more information about error occurred

Reply:
how would you suggest to map vyper SourceLocation to Solidity? 
Vyper: 
<img width="295" alt="image" src="https://user-images.githubusercontent.com/41516657/187642621-55a202d2-e75d-42f1-83d1-a25cfebc928e.png">


Solidity: <img width="268" alt="image" src="https://user-images.githubusercontent.com/41516657/187642551-2d6d4b2b-5393-4498-b336-28e2206da3ed.png">

I still think that this is something we don't want to

Reply:
My suggestion is to set `lineno` value to both `start` and `end`. But probably we could not do that now, I do not insist

Reply:
the thing is, start is not related to line number, it's first char in **one-line** code. so, it just makes no sense to put it there